GVFS/GVFS.Common/Git/GitObjects.cs

```diff
@@ -818,59 +840,97 @@ private LooseObjectToWrite GetLooseObjectDestination(string sha)
             }
         }
 
-        private Task<TempPrefetchPackAndIdx> StartPackIndexAsync(ITracer activity, PrefetchPacksDeserializer.PackAndIndex pack, string packName, string packTempPath, string idxName, string idxTempPath, Task packFlushTask)
+        private Task AddFinalizationTasks(TempPrefetchPackAndIdx currentOperation, Task previousPackTask)
+        {
+            currentOperation.ReadyTask = FinalizePackFileAsync(currentOperation, previousPackTask);
+            return currentOperation.ReadyTask;
+        }
+
+        private async Task FinalizePackFileAsync(TempPrefetchPackAndIdx currentOperation, Task previousPackTask)
+        {
+            await currentOperation.ReadyTask;
+            /* Before moving this pack and index from the temp folder to the live pack folder, create a marker
+             * file. This file is when getting the latest good pack timestamp to ignore the file.
+             * Delete it after the previous pack has finished.
+             * This lets us have the smaller, later pack files ready for use while the big initial
+             * file finishes indexing, while still making sure we start at the beginning next time 
+             * if this prefetch is interrupted.
+             */
+            string markerFilePath = Path.Combine(
+                this.Enlistment.GitPackRoot,
+                Path.ChangeExtension(currentOperation.PackName, GVFSConstants.InProgressPrefetchMarkerExtension));
+            this.fileSystem.OpenFileStream(markerFilePath, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.ReadWrite, true)
+                .Dispose();
+            string packDestination = Path.Combine(this.Enlistment.GitPackRoot, currentOperation.PackName);
+            if (!TryMovePackAndIdx(currentOperation.PackTempPath, packDestination, out Exception ex))
+            {
+                this.fileSystem.TryDeleteFile(markerFilePath);
+                throw ex;
+            }
+            await previousPackTask;

```

Comment:
To confirm my understanding, the `await previousPackTask;` will ensure that we always delete the marker files from oldest to newest? In other words, we'd never have the situation where we delete the marker file for a new pack file while the oldest is still be processed?

Reply:
Correct